### PR TITLE
[IN-2675] refactor flutter role

### PR DESCRIPTION
**Pull request by nagypeterjob**
### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [X] The tool I added is stable, and __does not require frequent updates__. _Preinstalled tools on the LTS stacks
  are not updated, so if the tool requires frequent updates you should handle the installation on-demand (see: [Install Any Additional Tool - DevCenter](https://bitrise-io.github.io/devcenter/tips-and-tricks/install-additional-tools/) for more information)_
- [X] I updated the corresponding tests if there were any.
- [X] I added a version report line to [`system_report.sh`](/system_report.sh) for the new tool(s) I added
- [X] I have added an entry to the CHANGELOG.md with a revisionID (the current date and the number of the change e.g.: 2019_10_11_1) and a brief description of the actual change
- [X] I have updated the revisionID in [bitrise_profile](roles/profiles/files/bitrise_profile) **BITRISE_OSX_STACK_REV_ID** with the one I just created in the CHANGELOG.md

### CODE REVIEW

**Review by tamas-bitrise — DISMISSED** [review]

File: roles/flutter/tasks/main.yml [hunk 1]
```diff
@@ -1,54 +1,65 @@
...
+  shell: bash -l -c "flutter --version"
+  ignore_errors: true
+  register: is_flutter_available
+  changed_when: false
+
+- name: Remove old flutter
+  become: true
+  file:
+    path: "/usr/local/flutter"
+    state: absent
```
Comment (fehersanyi-bitrise) [inline roles/flutter/tasks/main.yml:25]: when old flutter is available

Comment (fehersanyi-bitrise) [reply roles/flutter/tasks/main.yml:25]: @nagypeterjob if you don't when this, then the previous "is flutter available" thing is not needed

File: roles/flutter/tasks/main.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@
   file:
     path: "/usr/local/flutter"
     state: absent
+  when: is_flutter_available.rc != 0
```
Comment (fehersanyi-bitrise) [inline roles/flutter/tasks/main.yml:26]: I mean, if old flutter is available, delete is
it would be == 0 I believe

Comment (nagypeterjob) [reply roles/flutter/tasks/main.yml:26]: you are absolutely right :D

**Review by fehersanyi-bitrise — DISMISSED** [review]

**Review by tamas-bitrise — APPROVED** [review]

**Review by fehersanyi-bitrise — APPROVED** [review]
it should be cool now

**Pull request merged**